Podverse/MakeClipTimeViewController.swift

```diff
@@ -0,0 +1,276 @@
+            let playbackPosition = self.audioPlayer.progress
+            self.currentTime.text = Int64(playbackPosition).toMediaPlayerString()
+            let dur = self.audioPlayer.duration
+            self.duration.text = Int64(dur).toMediaPlayerString()
+            self.progress.value = Float(playbackPosition / dur)
+            
+            if let endTimePreview = self.endTimePreview {
+                if Int(self.audioPlayer.progress) >= endTimePreview {
+                    self.pvMediaPlayer.pause()
+                    self.endTimePreview = nil
+                }
+            }
+        }
+    }
+    
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        textField.backgroundColor = UIColor.yellow
+    }
+    
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        textField.backgroundColor = UIColor.white
+    }
+    
+    // Prevent select / paste menu options from appearing in UITextFields
+    override public func canPerformAction(_ action: Selector, withSender sender: Any?) -> Bool {
+        if startTimeInput.isFirstResponder || endTimeInput.isFirstResponder {
+            DispatchQueue.main.async {
+                (sender as? UIMenuController)?.setMenuVisible(false, animated: false)
+            }
+            return false
+        }
+        
+        return super.canPerformAction(action, withSender: sender)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        togglePlayIcon()
+        updateTime()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        navigationController?.interactivePopGestureRecognizer?.isEnabled = true
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.navigationItem.backBarButtonItem = UIBarButtonItem(title:"Back", style:.plain, target:nil, action:nil)
+        
+        setupTimer()
+        
+        addObservers()
+        
+        self.activityIndicator.startAnimating()
+        
+        updateTime()
+        
+        populatePlayerInfo()
+        
+        // prevent keyboard from displaying for startTimeInput and endTimeInput
+        self.startTimeInput.inputView = UIView()
+        self.endTimeInput.inputView = UIView()
+        
+        self.startTimeInput.text = PVTimeHelper.convertIntToHMSString(time: self.startTime)
+        
+        self.endTimeInput.becomeFirstResponder()
+        
+    }
+    
+    deinit {
+        removeObservers()
+        removeTimer()
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        
+        if segue.identifier == "Show Make Clip Title" {
+            
+            if let playerHistoryItem = playerHistoryItem {
+                let makeClipTitleViewController = segue.destination as! MakeClipTitleViewController
+                makeClipTitleViewController.playerHistoryItem = playerHistoryItem
+                
+                if let startTime = self.startTime {
```

Comment:
startTime is an optional in MakeClipTitleViewController so I don't think you need to unwrap